packages/dev/test-utils/src/dnd.ts

```diff
@@ -0,0 +1,113 @@
+import {act, fireEvent} from '@testing-library/react';
+// eslint-disable-next-line monorepo/no-internal-import
+import {DataTransfer, DragEvent} from '@react-aria/dnd/test/mocks';
+
+type DragPointerType = 'mouse' | 'touch' | 'pointer';
+
+type Delta = {x: number, y: number}
+
+function getElementCenter(element: Element) {
+  let {left, top, width, height} = element.getBoundingClientRect();
+  return {
+    x: left + width / 2,
+    y: top + height / 2
+  };
+}
+
+async function performDragAndDrop(source: Element, target: Element, delta: Delta, type: DragPointerType = 'mouse', steps = 1, duration = 0) {
```

Comment:
I forget exactly what values needed to be provided to perform a drop before/after a target, but it feels like a common enough use case to warrant a easy way to specify that drop operation. Maybe `delta` could be expanded to accept `on`/`before`/`after` and `delta` values as well? This all is assuming that the before/after drops simply were target's top/bottom coords which we could calculate for the user

Reply:
Great idea. Maybe `before`, `after`, `above` and `below` to cover all bases.

Reply:
on second thought, `toLeftOf` and `toRightOf` might be more clear than `before` and `after`.